sqlmesh/core/state_sync/base.py

```diff
@@ -248,7 +248,9 @@ def remove_interval(
     @abc.abstractmethod
     def promote(
         self, environment: Environment, no_gaps: bool = False
-    ) -> t.Tuple[t.List[SnapshotTableInfo], t.List[SnapshotTableInfo]]:
+    ) -> t.Tuple[
+        t.List[SnapshotTableInfo], t.Tuple[t.List[SnapshotTableInfo], EnvironmentNamingInfo]
```

Comment:
make this a triple instead of nested tuples

t.Tuple[a, b, c]

Reply:
I think the nested tuple makes the relationship clearer. The `EnvironmentNamingInfo` that is returned is specific to the `removed` snapshots. By pairing them together this relationship is more clear.

Reply:
it makes it harder to use, let's make it either a triple or make a simple data class to store the return type

Reply:
Data class works for me